### DISCUSSION TITLE
async support for ITransactionScope

### DISCUSSION THREAD

**Discussion by Andriy1024** [ChilliCream/hotchocolate#5629] [Feature Request]
Dear developers,
I faced the challenge to trigger some async work on the transaction completed event.
The current ITransactionScope's Complete method is not async compatible. It would be nice if you consider adding async support for this case.
Thank you!

The first option is to make 'Complete' methods async. In this case, it would be also nice to make it virtual in 'DefaultTransactionScope'.

```
public interface ITransactionScope : IDisposable
{
    /// <summary>
    /// Completes a transaction (commits or discards the changes).
    /// </summary>
    ValueTask CompleteAsync();
}
```

The second option is to use IAsyncDisposable

```
public interface ITransactionScope : IAsyncDisposable
{
...
}
```

**Comment by huer12** [comment]
We also need this feature. I would create a PR if it is ok for you. I would go for this if you agree:
```
public interface ITransactionScope : IDisposable, IAsyncDisposable
{
    ValueTask CompleteAsync();

    void Complete()
}
```

Would that be fine?

**Reply by huer12** [reply]
Added a PR https://github.com/ChilliCream/graphql-platform/pull/8227
